perun-base/src/main/java/cz/metacentrum/perun/core/impl/Privileges.java

```diff
@@ -0,0 +1,119 @@
+package cz.metacentrum.perun.core.impl;
+
+import cz.metacentrum.perun.core.api.Role;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static cz.metacentrum.perun.core.api.Role.FACILITYADMIN;
+import static cz.metacentrum.perun.core.api.Role.GROUPADMIN;
+import static cz.metacentrum.perun.core.api.Role.PERUNADMIN;
+import static cz.metacentrum.perun.core.api.Role.RESOURCEADMIN;
+import static cz.metacentrum.perun.core.api.Role.RESOURCESELFSERVICE;
+import static cz.metacentrum.perun.core.api.Role.SPONSOR;
+import static cz.metacentrum.perun.core.api.Role.TOPGROUPCREATOR;
+import static cz.metacentrum.perun.core.api.Role.VOADMIN;
+import static cz.metacentrum.perun.core.api.Role.VOOBSERVER;
+
+/**
+ * This class defines role setting privileges.
+ *
+ * This class contains information about which roles can set given role.
+ *
+ * @author Vojtech Sassmann <vojtech.sassmann@gmail.com>
+ */
+public class Privileges {
+
+	private static Map<Role, Set<Role>> rolePrivileges = new HashMap<>();
+
+	static {
+		initialize();
+	}
+
+	private static void initialize() {
+		role(FACILITYADMIN).canBeManagedBy(
+			FACILITYADMIN
+		);
+
+		role(GROUPADMIN).canBeManagedBy(
+			GROUPADMIN,
+			VOADMIN
+		);
+
+		role(PERUNADMIN).canBeManagedBy(
+			PERUNADMIN
+		);
+
+		role(RESOURCEADMIN).canBeManagedBy(
+			RESOURCEADMIN,
+			VOADMIN
+		);
+
+		role(RESOURCESELFSERVICE).canBeManagedBy(
+			RESOURCEADMIN,
+			VOADMIN
+		);
+
+		role(SPONSOR).canBeManagedBy(
+			VOADMIN
+		);
+
+		role(TOPGROUPCREATOR).canBeManagedBy(
+			VOADMIN
+		);
+
+		role(VOADMIN).canBeManagedBy(
+			VOADMIN
+		);
+
+		role(VOOBSERVER).canBeManagedBy(
+			VOOBSERVER,
+			VOADMIN
```

Comment:
Please support also newly added ``PERUNOBSERVER`` and ``SECURITYADMIN`` and ``CABINETADMIN``. All can be managed by themselves and PERUNADMIN.